async-query-core/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java

```diff
@@ -106,6 +106,13 @@ public DispatchQueryResponse submit(
         .resultIndex(dataSourceMetadata.getResultIndex())
         .datasourceName(dataSourceMetadata.getName())
         .jobType(JobType.INTERACTIVE)
+        .indexName(getIndexName(context))
```

Comment:
As mentioned below, we can't have name in all kinds of flint index queries? What is the indexName used for?

Reply:
DQS uses index name to populate their database to find the query where the index is created.